### PR TITLE
Common version

### PR DESCRIPTION
**Pull request by jmalkin**
* Bump versions for actions in workflows
* Define `version.cfg.in` as a single point to specify version
* Supporting changes to use the generated version

For dev builds, version will be major.minor.@DT@.@HHMM@ where they're expanded into the current UTC date and hour/minute. For releases, we'll specify the exact version number and the code to replace those variables will have nothing to do. But this should give us only one place to change the version.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -16,10 +16,24 @@
...
+string(TIMESTAMP HHMM %H%M UTC)
+configure_file(version.cfg.in ${CMAKE_CURRENT_SOURCE_DIR}/version.cfg @ONLY)
+file(STRINGS version.cfg BASE_VERSION)
+
 project(DataSketches
-        VERSION 3.6.0
+        VERSION ${BASE_VERSION}
         LANGUAGES CXX)
 
+#get_cmake_property(_variableNames VARIABLES)
```
Comment (AlexanderSaydakov) [inline CMakeLists.txt:29]: do we need to keep this commented-out section?

**Review by AlexanderSaydakov — APPROVED** [review]
looks good to me

File: CMakeLists.txt (hunk 1 above)
Comment (jmalkin) [reply CMakeLists.txt]: Removed -- thanks.

**Review by will-lauer — CHANGES_REQUESTED** [review]

File: .gitignore [hunk 2]
```diff
@@ -1,3 +1,7 @@
+# output of configured files
+version.cfg
+common/include/version.hpp
```
[the remark this answers is not in the document]
Comment (jmalkin) [reply .gitignore:3]: what about version.cfg?  and will it be picked up by make install if it's only in the build directory? i guess it should...

[the remark this answers is not in the document]
Comment (jmalkin) [reply .gitignore]: should be fixed now

File: common/include/version.hpp.in [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+constexpr int VERSION_MINOR {@DataSketches_VERSION_MINOR@};
+constexpr int VERSION_PATCH {@DataSketches_VERSION_PATCH@};
+constexpr int VERSION_TWEAK {@DataSketches_VERSION_TWEAK@};
+
+constexpr auto VERSION_STR = "@DataSketches_VERSION@";
+constexpr auto SOURCE_URL = "https://github.com/apache/datasketches-cpp";
+
+}
+
+#endif // _VERSION_HPP_
```
Comment (will-lauer) [inline common/include/version.hpp.in:36]: missing newline?

**Review by will-lauer — APPROVED** [review]

File: common/include/version.hpp.in (hunk 3 above)
Comment (jmalkin) [reply common/include/version.hpp.in]: fix, thx

**Pull request merged**